src/containerapp/azext_containerapp/containerapp_env_storage_decorator.py

```diff
@@ -0,0 +1,135 @@
+        super().__init__(cmd, client, raw_parameters, models)
+        self.managed_environment_storage_def = ManagedEnvironmentStorageProperties
+        self.storage_type = self.get_argument_storage_type()
+        self.azure_file_account_name = self.get_argument_azure_file_account_name()
+        self.azure_file_share_name = self.get_argument_azure_file_share_name()
+        self.azure_file_account_key = self.get_argument_azure_file_account_key()
+        self.server = self.get_argument_server()
+        self.access_mode = self.get_argument_access_mode()
+
+    def get_argument_storage_name(self):
+        return self.get_param("storage_name")
+
+    def get_argument_storage_type(self):
+        return self.get_param("storage_type")
+
+    def get_argument_azure_file_account_name(self):
+        return self.get_param("azure_file_account_name")
+
+    def get_argument_azure_file_share_name(self):
+        return self.get_param("azure_file_share_name")
+
+    def get_argument_azure_file_account_key(self):
+        return self.get_param("azure_file_account_key")
+
+    def get_argument_server(self):
+        return self.get_param("server")
+
+    def get_argument_access_mode(self):
+        return self.get_param("access_mode")
+
+    def construct_payload(self):
+        if not self.storage_type or self.storage_type.lower() == AZURE_FILE_STORAGE_TYPE.lower():
+            storage_def = AzureFileProperties
+            storage_def["accountKey"] = self.azure_file_account_key
+            storage_def["accountName"] = self.azure_file_account_name
+            storage_def["shareName"] = self.azure_file_share_name
+            storage_def["accessMode"] = self.access_mode
+
+            self.managed_environment_storage_def["properties"] = {}
+            self.managed_environment_storage_def["properties"]["azureFile"] = storage_def
+        elif self.storage_type.lower() == NFS_AZURE_FILE_STORAGE_TYPE.lower():
+            storage_def = NfsAzureFileProperties
+            storage_def["server"] = self.server
+            storage_def["shareName"] = self.azure_file_share_name
+            storage_def["accessMode"] = self.access_mode
+            self.managed_environment_storage_def["properties"] = {}
+            self.managed_environment_storage_def["properties"]["nfsAzureFile"] = storage_def
+
+    def validate_arguments(self):
+        import json
+        if not self.storage_type or self.storage_type.lower() == AZURE_FILE_STORAGE_TYPE:
+            if len(self.azure_file_share_name) == 0 or len(self.azure_file_account_name) == 0 or len(
+                    self.azure_file_account_key) == 0 or len(self.access_mode) == 0:
+                raise RequiredArgumentMissingError(
+                    "Storage account name, key, share name and access_mode must be provided for AzureFile storage type")
+            if len(self.azure_file_share_name) < 3:
+                raise ValidationError("File share name must be longer than 2 characters.")
```

Comment:
Same as above, please use the parameter name:
`raise ValidationError("File share name with --azure-file-share-name/--file-share/-f must be longer than 2 characters.")`